openbas-api/src/main/java/io/openbas/utils/CopyObjectListUtils.java

```diff
@@ -33,4 +43,29 @@ public static <T extends Base, C extends Collection<T>> C copyCollection(@NotNul
         return destinations;
     }
 
+    public static <T, C> T copyObjectWithoutId(C origin, Class<T> targetClass) {
+        try {
+            T target = targetClass.getDeclaredConstructor().newInstance();
+
+            // Get all declared fields from the source object
+            Field[] fields = origin.getClass().getDeclaredFields();
+
+            for (Field field : fields) {
+                field.setAccessible(true);
+
+                // Skip the 'id' field
+                if (field.getName().equals("id")) {
```

Comment:
Using reflection, would it be possible to check if the field has the jakarta.persistence.Id annotation ? It would prevent the use case where an Id field is not named 'id' (imagine it's importerId instead for instance).